src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -2322,7 +2322,7 @@ bool DiscoveryDataBase::from_json(
             }
 
             // Add Participant
-            auto wit = writers_.insert(std::make_pair(guid_aux, dei));
+            writers_.insert(std::make_pair(guid_aux, dei));
```

Comment:
That's a tricky one, as the variable is indeed used in line 2346, which is a `logInfo` message that is not compiled unless `INTERNAL_DEBUG` is set or `CMAKE_BUILD_TYPE` is set to `Debug`. Our CI builds with `INTERNAL_DEBUG` to verify that all `logInfo` are correct, and thus we don't get the warning. The best is probably to remove the part of the log that uses it.

Reply:
What do you think about just putting in a `static_cast<void>`, like my most recent commit? It suppresses the warning, but still works for the log?